### PR TITLE
Enable navmesh

### PR DESCRIPTION
**Pull request by PatriceJiang**

### CODE REVIEW

File: cocos/2d/CCScene.cpp [hunk 1]
```diff
@@ -254,8 +254,6 @@ void Scene::render(Renderer* renderer, const Mat4& eyeTransform, const Mat4* eye
         _physics3DWorld->debugDraw(renderer);
         renderer->render();
 
-        physics3dDebugCamera->restore();
-
```
Comment (minggo) [inline cocos/2d/CCScene.cpp:258]: why remote it?

File: cocos/navmesh/CCNavMeshDebugDraw.h [hunk 2]
```diff
@@ -80,29 +89,40 @@ class NavMeshDebugDraw : public duDebugDraw
...
-        GLenum type;
-        bool depthMask;
-        unsigned short start;
-        unsigned short end;
-        float size;
+        backend::PrimitiveType  type;
+        bool                    depthMask;
+        unsigned short          start;
+        unsigned short          end;
+        float                   size;
```
Comment (minggo) [inline cocos/navmesh/CCNavMeshDebugDraw.h]: please set default values

File: cocos/2d/CCScene.cpp (hunk 1 above)
Comment (PatriceJiang) [reply cocos/2d/CCScene.cpp:258]: the method no longer exists and not referenced

File: cocos/navmesh/CCNavMeshDebugDraw.h (hunk 2 above)
Comment (PatriceJiang) [reply cocos/navmesh/CCNavMeshDebugDraw.h]: ok

File: cocos/navmesh/CCNavMeshDebugDraw.h [hunk 3]
```diff
@@ -80,29 +89,40 @@ class NavMeshDebugDraw : public duDebugDraw
...
-    GLenum _primitiveType;
-    bool _currentDepthMask;
-    GLuint _vbo;
-    bool _dirtyBuffer;
+    Primitive             * _currentPrimitive   = nullptr;
+    backend::ProgramState * _programState       = nullptr;
+    bool _currentDepthMask                      = true;
+    bool _dirtyBuffer                           = true;
+    backend::PrimitiveType  _primitiveType      = backend::PrimitiveType::POINT;
+    backend::Buffer       * _vertexBuffer       = nullptr;
```
Comment (minggo) [inline cocos/navmesh/CCNavMeshDebugDraw.h:104]: Don't have to initialize these variables in constructor initialization list.

**Review by minggo — APPROVED** [review]

**Pull request merged**